lib/uring/uring.mli

```diff
@@ -67,6 +67,15 @@ val noop : 'a t -> 'a -> 'a job option
 (** [noop t d] submits a no-op operation to uring [t]. The user data [d] will be
     returned by {!wait} or {!peek} upon completion. *)
 
+val timeout: 'a t -> [`Boottime | `Realtime | `Timeout_abs] -> int64 -> 'a -> 'a job option
```

Comment:
```suggestion
val timeout: 'a t -> [`Boottime | `Realtime] -> [`Relative|`Absolute] -> int64 -> 'a -> 'a job option
```
`IORING_TIMEOUT_ABS` is a flag, not a clock.

Reply:
Not sure what is happening but when I use `IORING_TIMEOUT_ABS` both as a flag or the only clock source, the timeout is never armed, i.e. `flags |= IORING_TIMEOUT_ABS`. My github and google-fu for IORING_TIMEOUT_ABS doesn't bring that much clarity. I think we should drop this flag for now and add it in the future PR. I don't immediately have need of this flag for now.